horovod/torch/ready_event.cc

```diff
@@ -57,7 +56,6 @@ TorchReadyEvent::TorchReadyEvent(int device) : device_(device) {
   }
   auto stream = THCState_getCurrentStreamOnDevice(state, device_);
   THCudaCheck(cudaEventRecord(cuda_event_, stream));
-  THCudaCheck(cudaSetDevice(restoreDevice));
```

Comment:
for streams and events, you can use the aten Stream API now. For example: `at::cuda::getCurrentCUDAStream()`. See https://github.com/pytorch/pytorch/pull/9435 for reference.

Events are also going to be exposed on ATen, PR is out: https://github.com/pytorch/pytorch/pull/9726